app/models/git_repository.rb

```diff
@@ -128,15 +154,19 @@ def refresh_tags
 
   def worktree
     @worktree ||= begin
-      clone_repo unless Dir.exist?(directory_name)
-      fetch_worktree
+      git_transaction do
```

Comment:
Unsure if this is needed... I don't think `GitRepository#worktree` is ever accessed without `GitRepository#with_worktree`, but maybe I am missing something.

If nothing else, it at least tests that the "top level only" transaction thing works (the test suite should exercise this cod) from `EmbeddedAnsible::AutomationManager::ConfigurationScriptSource`).

Reply:
I think only the clone line + directory check needs the lock around it.  I'd be ok if the Dir.exist? check lived inside clone_repo method so that you can encapsulate the lock in that one method.